src/service/service.ts

```diff
@@ -338,13 +343,17 @@ export class Discv5 extends (EventEmitter as { new (): Discv5EventEmitter }) {
       }
     });
   }
-
   /**
    * Send TALKREQ message to dstId and returns response
    */
-  public async sendTalkReq(dstId: string, payload: Buffer, protocol: string | Uint8Array): Promise<Buffer> {
+  public async sendTalkReq(
```

Comment:
I think we can change the interface here to be more aligned with what we need to send an Rpc request/response. (Also replace `Buffer` with `Uint8Array`.) I'm thinking something like this:
```ts
public async sendTalkReq(dst: ENR | Multiaddr, payload: Uint8Array, protocol: string | Uint8Array): Promise<Uint8Array> {
  return await new Promise((resolve, reject) => {
    this.sendRpcRequest({
      contact: createNodeContact(dst),
      request: createTalkRequestMessage(payload, protocol),
      callback: (err: RequestErrorType | null, res: Uint8Array | null): void => {
        if (err !== null) {
          reject(err);
          return;
        }
        resolve(res as Uint8Array);
      },
    });
  });
}
```

We can also expose `findEnr` as a public method, so anyone who wants something like the existing situation can find the enr themselves.
```ts
const enr = discv5.findEnr(dstId);
if (!enr) {
  throw ...;
}
const resp = await discv5.sendTalkReq(enr, payload, protocol);
```

And then we can have a similar interface/implementation for `sendTalkResp`

Reply:
So make the consuming app provide the ENR up front and remove the ENR lookup entirely from within `sendTalkReq` (and friends)?

Reply:
Yeah!

Reply:
Yeah, it is. If we want the other stuff in the current version, we can break this up into another PR.
Afaik, lodestar and ultralight are the only users of this library, moving quickly to another version shouldn't be an issue.